sql/transactions/transactions.go

```diff
@@ -96,7 +96,9 @@ func HasBlockTX(db sql.Executor, bid types.BlockID, tid types.TransactionID) (bo
 func UpdateIfBetter(db sql.Executor, tid types.TransactionID, lid types.LayerID, bid types.BlockID) (int, error) {
 	rows, err := db.Exec(`
 		update transactions set layer = ?5, block = ?6
-		where id = ?1 and applied = ?2 and (layer = ?3 or layer > ?5 or (layer = ?5 and block = ?4)) returning id`,
+		where id = ?1 and applied = ?2 and
+		(layer is null or layer = ?3 or layer > ?5 or (layer = ?5 and (block = ?4 or block is null)))
```

Comment:
having so many or-s in request should probably make all indexes skipped to use. Or we don't much care about request performance in this case?

Reply:
hmmmm.... i care about performance. what do you mean by `make all indexes skipped to use`?

`id` is the primary key, and should be the only index used in this query.

Reply:
ah, okay, sorry :) what I meant is indexes never used for `or` but as long as `id` is enough to find a record - should be ok, cheers :)

Reply:
thanks for the review. yeah. the primary key already narrow it down to 0 or 1 record. we can't do better that. more index will not help with the value comparisons on fields. this query essentially does 0 or 1 row update.

please do keep reminding in the future about index. it's always an important question to ask when writing sql queries.